## Christian Huerta
Real-Time Big Data Analytics<br>
Project: Marlins vs. Dodgers Game Highlights (4/29/25)<br>

### Installation: 

In [14]:
# pip install youtube-transcript-api textblob

### Import Libaries:

In [15]:
from youtube_transcript_api import YouTubeTranscriptApi
from textblob import TextBlob
import matplotlib.pyplot as plt

### URL YouTube Video ID:

In [ ]:
# URL: https://www.youtube.com/watch?v=aeKt4VHi7XM&ab_channel=MLB
video_id = "aeKt4VHi7XM&ab_channel=MLB"

### Extracting the Transcript:

In [ ]:
# Fetch transcript
transcript = YouTubeTranscriptApi.get_transcript(video_id)

### Joining Transcript Segments into a Single Text Block:

In [12]:
# Join all text segments
video_text = " ".join([segment['text'] for segment in transcript])
print(video_text[:1000])

Dodger bullpen. They throw more innings than any bullpen in baseball. They deserve a break. They're not gonna get one here tonight. It's a bullpen game, bullpen game. And remember we had one similar to this against the Cubs and it didn't go well, Jack Dryer get the ball tonight. He has started in his baseball life before Xavier Edwards will lead things off. Xavier Edwards was one for five game one last night that is lined towards the corner in right field that's going to fall for a base hit. It's going to get past Hernandez and roll all the way to the wall. Edwards can really run on his way to third. They'll stop him there with a leadoff triple to start the ball game. Hernandez plays right field in this stadium. A little surprised how he played that ball. Augustin ramirez strike three painted a fastball on the inside corner with all don't live. Here's Dan Myers who came off the bench last night and went two for two hit his first career grand slam and there's a swing of a line drive to 

### Performing Sentiment Analysis with TextBlob:

This block of code uses `TextBlob` to analyze the **overall sentiment** of the transcript text. 

- TextBlob(video_text) creates a TextBlob object, which allows for various text processing operations.

- blob.sentiment returns a tuple with two values:

- Polarity: a value between -1.0 and 1.0 indicating sentiment direction.

  -  '> 0 → positive sentiment

  -  < 0 → negative sentiment

  - ≈ 0 → neutral sentiment

- Subjectivity: a value between 0.0 and 1.0 showing how opinion-based the text is.

  - 0.0 → very objective (factual)

  - 1.0 → very subjective (personal opinions)

These values help us understand whether the speaker expresses strong emotions or remains neutral/factual in tone.

### Performing Sentiment Analysis with TextBlob:

In [13]:
blob = TextBlob(video_text)
print("Sentiment Polarity:", blob.sentiment.polarity)
print("Sentiment Subjectivity:", blob.sentiment.subjectivity)

Sentiment Polarity: 0.029379169849758066
Sentiment Subjectivity: 0.3711311858076565


## 🎯 Sentiment Analysis Interpretation:

**Video Title:** *Marlins vs. Dodgers Game Highlights (4/29/25)*  
**Source:** [MLB YouTube Channel](https://www.youtube.com/watch?v=aeKt4VHi7XM)

---

### 📊 Result Recap
- **Sentiment Polarity:** `0.0294`  
- **Sentiment Subjectivity:** `0.3711`

---

### ⚖️ Polarity Interpretation
- The **polarity score** is slightly above **zero**, indicating a **mildly positive** tone.
- This suggests the commentary likely highlights **positive aspects** of the game, such as standout performances or exciting moments, without significant negative criticism.

---

### 🧠 Subjectivity Interpretation
- The **subjectivity score** of approximately `0.37` implies the content is **more objective than subjective**.
- This is typical for **sports highlight videos**, which focus on factual play-by-play coverage with occasional commentary.

---

### 🧾 Conclusion
- The video's tone is **informative and neutral**, emphasizing the game's events without strong emotional language.
- The sentiment reflects a **balanced and factual** presentation, suitable for viewers seeking a straightforward recap of the game's highlights.
